### PR TITLE
Make Callbacker conform to SignalProvider

### PR DESCRIPTION
**Pull request by niil-ohlin**

### CODE REVIEW

File: Flow/Signal+Listeners.swift [hunk 1]
```diff
@@ -103,6 +110,21 @@ public extension SignalProvider where Kind == ReadWrite {
 
         return bag
     }
+
+    /// Start listening to `self` and add a callback to `callbacker` which updates `self.value`
+    /// - Parameter isSame: Are two values the same. Used to avoid infinite recursion.
+    /// - Returns: A disposable that will stop listening on values when being disposed.
+    func bidirectionallyBindTo(_ callbacker: Callbacker<Value>, isSame: @escaping (Value, Value) -> Bool) -> Disposable {
+        let bag = DisposeBag()
+        bag += callbacker.addCallback { (value: Value) in
```
Comment (mansbernhardt) [inline Flow/Signal+Listeners.swift]: As call backer is a signal provider, should not something like this work?

```swift
bag += callbacker.distinct().bindTo(signal)
```

File: Flow/Signal+Listeners.swift [hunk 2]
```diff
@@ -87,6 +87,13 @@ public extension SignalProvider {
             value[keyPath: keyPath] = $0
         }
     }
+
+    /// Start listening on values and call `callbacker.callAll(with:)` when receiving a value.
+    /// - Returns: A disposable that will stop listening on values when being disposed.
+    /// - Note: equivalent to `onValue(callbacker.callAll(with:))`
+    func bindTo(on scheduler: Scheduler = .current, _ callbacker: Callbacker<Value>) -> Disposable {
+        return onValue(on: scheduler , callbacker.callAll(with:))
+    }
```
Comment (mansbernhardt) [inline Flow/Signal+Listeners.swift]: I think this is probably better solved by:
```swift
signal.bindTo(callbacker.callAll)
```

File: Flow/Signal+Listeners.swift [hunk 3]
```diff
@@ -103,6 +110,21 @@ public extension SignalProvider where Kind == ReadWrite {
 
         return bag
     }
+
+    /// Start listening to `self` and add a callback to `callbacker` which updates `self.value`
+    /// - Parameter isSame: Are two values the same. Used to avoid infinite recursion.
+    /// - Returns: A disposable that will stop listening on values when being disposed.
+    func bidirectionallyBindTo(_ callbacker: Callbacker<Value>, isSame: @escaping (Value, Value) -> Bool) -> Disposable {
```
Comment (mansbernhardt) [inline Flow/Signal+Listeners.swift]: I think perhaps this is the only method the we can motivate as it's a bit complicated to write. On the other hand, how often do we see this being used? Do you have a use case?

File: Flow/Callbacker.swift [hunk 4]
```diff
@@ -102,3 +102,9 @@ public extension Callbacker where Value == () {
         callAll(with: ())
     }
 }
+
+extension Callbacker: SignalProvider {
+    public var providedSignal: Signal<Value> {
+        return Signal(callbacker: self)
+    }
+}
```
Comment (mansbernhardt) [inline Flow/Callbacker.swift]: Can't decide I like this or not. It sort of make sense if you are doing transforms on the callbacker, but the most common use case is probable to return it as a signal so we have 

```swift
return Signal(callbacker: callbacker)
// vs
return callbacker.providedSignal
```

But it of course gets better once doing transforms:

```swift
return Signal(callbacker: callbacker).map { ... }
// vs
return callbacker.map { ... }
```

or perhaps when making it readable:

```swift
return Signal(capturing: self.value, callbacker: callbacker)
// vs
return callbacker.readable(capturing: self.value) // A bit less efficient
```

So overall I think I give it a thumbs up, so the user can choose on case by case.

File: Flow/Signal+Listeners.swift (hunk 3 above)
Comment (niil-ohlin) [reply Flow/Signal+Listeners.swift]: Not really. If we find it useful maybe we can add it later.

**Comment by niil-ohlin** [comment]
I removed the `bidirectionalBindTo` for the callbackers. @mansbernhardt what do you think of this as a start? I think that this would be very useful for us.

**Pull request merged**